### PR TITLE
[#501] Add build for Domainmodel example

### PR DESCRIPTION
**Pull request by kthoms**
Adds build definition for the Domainmodel example.

Test:
1. run `mvn package` on project org.eclipse.xtext.xtext.ui.examples. This will bundle the domainmodel projects
2. In runtime Eclipse, create the Domainmodel example with the wizard
3. Execute `buildDomainmodel.launch` in the new releng project.

screenshot 37
 

- add project 'org.eclipse.xtext.example.domainmodel.releng'
  - contains parent pom.xml
  - contains submodule 'targetplatform' defining the build target
    (avoiding yet another project)
- register new project to build.xml and plugin.xml
- added pom.xml files for all domainmodel projects
- added aggregator pom.xml in folder 'projects'

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.ui.examples/build.xml [hunk 1]
```diff
@@ -60,6 +60,7 @@
 		<zip_example subdir="domainmodel"    name="org.eclipse.xtext.example.domainmodel.tests" />
 		<zip_example subdir="domainmodel"    name="org.eclipse.xtext.example.domainmodel.ui" />
 		<zip_example subdir="domainmodel"    name="org.eclipse.xtext.example.domainmodel.ui.tests" />
+		<zip_example subdir="domainmodel"    name="org.eclipse.xtext.example.domainmodel.releng" />
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.ui.examples/build.xml]: usual naming convention is .parent

Comment (kthoms) [reply org.eclipse.xtext.xtext.ui.examples/build.xml]: it is more than just the parent, it contains also the target platform. I don't like to have too much top-level projects which contain only releng related stuff. Also for repository and feature project.

Comment (kthoms) [reply org.eclipse.xtext.xtext.ui.examples/build.xml]: I know it differs from the structure we usually produce with the wizard. But the EMF example wizard can only install separate projects. Better would be to Zip the whole example as one archive and place the parent POM in the root folder.
I had to create a separate project for the releng stuff, thus decided to call it "releng".

Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.examples/build.xml]: dont get you. id like to have the name a.b.c.parent as we do it at different places and which is  a common pattern

Comment (kthoms) [reply org.eclipse.xtext.xtext.ui.examples/build.xml]: The parent project created with the wizard is physically the root folder of the other projects, which is common Maven style.
We can't do this with the wizard, we have to produce a sibling project that contains the parent pom.xml. Of course, we could name that "parent".

That's the project structure created by the wizard:
screenshot 40

That's the proposed Domainmodel example project structure:
screenshot 41

If we would want to use the typical structure, we can't go with the EMF example project anymore. Or we just install the "root" project and have to import the nested projects afterwards.

Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.examples/build.xml]: ok go for releng

**Pull request merged**